### PR TITLE
Move link props CRUD operations out of TER

### PR DESCRIPTION
**Pull request by surabujin**
To be able to remove TER from production environment move out from it all endpoints used in production code.

### CODE REVIEW

File: services/src/messaging/src/main/java/org/openkilda/messaging/nbtopology/response/LinkPropsData.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import org.openkilda.messaging.info.InfoData;
+import org.openkilda.messaging.model.NetworkEndpoint;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Getter;
+
+import java.util.Map;
+
+@Getter
+public class LinkPropsData extends InfoData {
```
Comment (surabujin) [inline services/src/messaging/src/main/java/org/openkilda/messaging/nbtopology/response/LinkPropsData.java]: @nikitacherevko you should use LinkProps model object wrapped into some transport layer(`InfoData` in this case).

File: services/src/messaging/src/main/java/org/openkilda/messaging/model/AbstractNetworkEndpoint.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    @JsonProperty("port-id")
+    private final Integer portNumber;
+
+    @JsonCreator
+    public AbstractNetworkEndpoint(String datapath, Integer portNumber) {
+        this.datapath = datapath;
+        this.portNumber = portNumber;
+
+        validate();
```
Comment (sergii-iakovenko) [inline services/src/messaging/src/main/java/org/openkilda/messaging/model/AbstractNetworkEndpoint.java]: This is bad practice. "Constructor must not invoke overridable methods" (Item 17: Design and document for inheritance or else prohibit it. Effective Java, 2nd edition. J. Bloch)

File: services/src/messaging/src/main/java/org/openkilda/messaging/nbtopology/response/LinkPropsData.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import org.openkilda.messaging.model.LinkProps;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jdk.nashorn.internal.objects.annotations.Property;
+import lombok.Value;
+
+@Value
+public class LinkPropsData extends InfoData {
+
+    @Property
```
Comment (sergii-iakovenko) [inline services/src/messaging/src/main/java/org/openkilda/messaging/nbtopology/response/LinkPropsData.java]: Shouldn't it be @JsonProperty?

File: services/src/messaging/src/test/java/org/openkilda/messaging/model/LinkPropsMaskTest.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+import org.openkilda.messaging.StringSerializer;
+import org.openkilda.messaging.Utils;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+
+public class LinkPropsMaskTest implements StringSerializer {
```
Comment (sergii-iakovenko) [inline services/src/messaging/src/test/java/org/openkilda/messaging/model/LinkPropsMaskTest.java]: StringSerializer is used here only for keeping serialized value. I believe local var is easier and fits the purpose better.

File: services/src/northbound/src/main/java/org/openkilda/northbound/utils/CorrelationIdUuidFactory.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+package org.openkilda.northbound.utils;
+
+import org.springframework.stereotype.Component;
+
+import java.util.UUID;
+
+@Component
+public class CorrelationIdUuidFactory implements CorrelationIdFactory {
+    private String lastId = null;
```
Comment (sergii-iakovenko) [inline services/src/northbound/src/main/java/org/openkilda/northbound/utils/CorrelationIdUuidFactory.java]: What's the purpose of state in spring singleton component? This is thread unsafe.

File: services/src/northbound/src/test/java/org/openkilda/northbound/utils/CorrelationIdPredictableFactory.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.northbound.utils;
+
+public class CorrelationIdPredictableFactory extends CorrelationIdUuidFactory {
+    private int counter = 0;
```
Comment (sergii-iakovenko) [inline services/src/northbound/src/test/java/org/openkilda/northbound/utils/CorrelationIdPredictableFactory.java]: What's the purpose of state in spring singleton component? This is thread unsafe.

**Comment by nikitacherevko** [comment]
Closes #864

**Pull request merged**